vita3k/mem/src/mem.cpp

```diff
@@ -43,7 +43,14 @@ static void delete_memory(uint8_t *memory) {
 }
 
 static void alloc_inner(MemState &state, Address address, size_t page_count, Allocated::iterator block, const char *name) {
-    uint8_t *const memory = &state.memory[address];
+    Address alignment;
+#ifdef WIN32
+    alignment = 4096;
+#else
+    alignment = getpagesize();
+#endif
```

Comment:
The MemState should have the correct page size already.

Reply:
Does `state.page_size` have the ps vita page size or the host's page size? I thought it was the vita's...

Reply:
At the moment it's host, but after @pent0's comment below, `state.page_size` should probably be `max(4096, host page size)` instead of just the host page size.

Reply:
Yep, that's should be right.

Reply:
Wait a minute, wouldn't that be essentially the same as locking it to 4096?

If host size is under 4096, host size will be set to 4096.
If host size is above 4096, this assert in MemState init will go off.
```
assert(state.page_size >= 4096); // Limit imposed by Unicorn.
```

Reply:
Well, what if there is Unicorn 2.0 and it doesn't limit page size to 4096? Seriously don't throw away the step!

Reply:
I'll do it anyway 😅

Reply:
Merci boku naruto uchiha